examples/autoscaler-wasm/model.go

```diff
@@ -0,0 +1,139 @@
+	Count             int64 `json:"count"`
+	Capacity          int64 `json:"capacity"`
+}
+
+// AggregatedListStatus stores total and allocated List tracking values
+type AggregatedListStatus struct {
+	AllocatedCount    int64 `json:"allocatedCount"`
+	AllocatedCapacity int64 `json:"allocatedCapacity"`
+	Count             int64 `json:"count"`
+	Capacity          int64 `json:"capacity"`
+}
+
+// FleetStatus is the status of a Fleet
+type FleetStatus struct {
+	// Replicas the total number of current GameServer replicas
+	Replicas int32 `json:"replicas"`
+	// ReadyReplicas are the number of Ready GameServer replicas
+	ReadyReplicas int32 `json:"readyReplicas"`
+	// ReservedReplicas are the total number of Reserved GameServer replicas in this fleet.
+	// Reserved instances won't be deleted on scale down, but won't cause an autoscaler to scale up.
+	ReservedReplicas int32 `json:"reservedReplicas"`
+	// AllocatedReplicas are the number of Allocated GameServer replicas
+	AllocatedReplicas int32 `json:"allocatedReplicas"`
+	// [Stage:Alpha]
+	// [FeatureFlag:PlayerTracking]
+	// Players are the current total player capacity and count for this Fleet
+	// +optional
+	Players *AggregatedPlayerStatus `json:"players,omitempty"`
+	// (Beta, CountsAndLists feature flag) Counters provides aggregated Counter capacity and Counter
+	// count for this Fleet.
+	// +optional
+	Counters map[string]AggregatedCounterStatus `json:"counters,omitempty"`
+	// (Beta, CountsAndLists feature flag) Lists provides aggregated List capacityv and List values
+	// for this Fleet.
+	// +optional
+	Lists map[string]AggregatedListStatus `json:"lists,omitempty"`
+}
+
+// FleetAutoscalerPolicyType is the policy for autoscaling
+// for a given Fleet
+type FleetAutoscalerPolicyType string
+
+// FleetAutoscalerStatus defines the current status of a FleetAutoscaler
+type FleetAutoscalerStatus struct {
+	// CurrentReplicas is the current number of gameserver replicas
+	// of the fleet managed by this autoscaler, as last seen by the autoscaler
+	CurrentReplicas int32 `json:"currentReplicas"`
+
+	// DesiredReplicas is the desired number of gameserver replicas
+	// of the fleet managed by this autoscaler, as last calculated by the autoscaler
+	DesiredReplicas int32 `json:"desiredReplicas"`
+
+	// lastScaleTime is the last time the FleetAutoscaler scaled the attached fleet,
+	// +optional
+	LastScaleTime *time.Time `json:"lastScaleTime"`
+
+	// AbleToScale indicates that we can access the target fleet
+	AbleToScale bool `json:"ableToScale"`
+
+	// ScalingLimited indicates that the calculated scale would be above or below the range
+	// defined by MinReplicas and MaxReplicas, and has thus been capped.
+	ScalingLimited bool `json:"scalingLimited"`
+
+	// LastAppliedPolicy is the ID of the last applied policy in the ChainPolicy.
+	// Used to track policy transitions for logging purposes.
+	LastAppliedPolicy FleetAutoscalerPolicyType `json:"lastAppliedPolicy"`
+}
```

Comment:
This doesn't seems to be used, we might want to delete it ?

Reply:
Also well spotted! Deleted!